modules/fedimint-mint-client/src/backup/recovery.rs

```diff
@@ -367,6 +367,11 @@ impl MintRecoveryState {
             )
             .is_none());
 
+        assert!(
+            self.pending_nonces.len() >= (self.gap_limit as usize) * self.tbs_pks.count_tiers(),
```

Comment:
We remove nonces we've seen already, so this might fail, no?

Reply:
Hmmmm... For every removed one we should add another one ...

Reply:
Yeah, but initially we start with an empty one and fill it up one by one, so this is going to immediately fail.

Reply:
Ok, then not a viable assert unfortunately.